tests/unit/json_lib/parser_test.py

```diff
@@ -273,6 +279,98 @@ def test_parse_from_bytes(self):
         self.assertEquals(x.get("c"), True)
         self.assertEquals(x.get("d"), "Hello")
 
+    def test_unexpected_object_end(self):
+        json_data = """
+            {
+                api_key: ",key."
+                field1: 1,
+            }
+                filed2: 2
+            }
+        """
+
+        with pytest.raises(JsonParseException) as err_info:
+            JsonParser.parse(json_data)
+
+        assert "Expecting EOF, got 'f'" in str(err_info.value)
+
+    def test_comma_at_the_end(self):
+        json_data = """
+            {
+                api_key: "<key>"
+                field1: 1,
+                field2: 2
+            },
+        """
+        with pytest.raises(JsonParseException) as err_info:
+            JsonParser.parse(json_data)
+
+        assert "Expecting EOF, got ','" in str(err_info.value)
+
+    def test_commented_start_of_the_object(self):
+        """
+        Since the JSON parser supports comments, there may be the case where the beginning of the object is
+        (accidentally) commented, so the internal fields of that object are parsed as fields of the external object.
+        """
+        json_data = """
+            {
+              api_key: ",key."
+              // server_attributes: {       // the beginning of the nested object is commented.
+                 serverHost: "HOST",        // the 'serverHost' field is wrongly parsed as a filed of outer object.
+              }                             // this brackets are treated as end of the whole JSON document.
+
+              // Everything that remains is just ignored, that's bad.
+              monitors: [
+                {
+                   module:  "my_module",
+                },
+              ]
+            }
+"""
+        with pytest.raises(JsonParseException) as err_info:
+            JsonParser.parse(json_data)
+
+        assert "Expecting EOF, got 'm'" in str(err_info.value)
```

Comment:
The error message is not the most user-friendly, but I imagine if we wanted to provide more context, we would need to substantially change the parsing code so we have access to more context and not just the next / current character.

On that note, does the error class / message already include the line number where the error occured?

Reply:
Changed error message to `"Expecting end of the parsed document at the character '105', got character 'f' instead"`, WDYT?

Reply:
@Kami Line number is also shown now. I've also tried to add the column number, but I guess it's better to do that separately since it requires more refactoring.